### PR TITLE
vouch-proxy: init at 0.23.1

### PR DESCRIPTION
**Pull request by richardlarocque**
Initializes a package for vouch-proxy.  Its developers describe it as
"an SSO and OAuth / OIDC login solution for Nginx using the auth_request
module".

Upstream project is here: https://github.com/vouch/vouch-proxy.

###### Motivation for this change

https://github.com/NixOS/nixpkgs/issues/90728

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by richardlarocque** [comment]
I don't have a lot of time to work on this, so I'm explicitly not stepping up to be a maintainer here.  I may not be very responsive to PR comments, either.  Apologies in advance.

Still, I figure this code drop is better than keeping the work to myself, and so here it is.  If someone else wants to take this PR and run with it, that would be much appreciated.  Feel free to remix the code as you see fit.

**Comment by richardlarocque** [comment]
[vouch-module.nix.txt](https://github.com/NixOS/nixpkgs/files/6259739/vouch-module.nix.txt)

Related: I built a rudimentary NixOS module for vouch-proxy, too.  I don't recommend using it as is.  A better module would abstract away the details of vouch-proxy's YAML configs.  A much better module would provide more help in configuring nginx to work with it.  

Once again, this is a code drop without promise or maintenance of improvement.  I'm sorry I couldn't offer more, and I hope it's better than nothing.

File: pkgs/servers/vouch-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/vouch-proxy/default.nix:7]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/servers/vouch-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/vouch-proxy/default.nix:1]:
```suggestion
{ lib, buildGoModule, makeWrapper, fetchFromGitHub }:
```

File: pkgs/servers/vouch-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/vouch-proxy/default.nix:3]:
```suggestion
buildGoModule rec {
```

File: pkgs/servers/vouch-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/vouch-proxy/default.nix:9]:
```suggestion
  src = fetchFromGitHub {
```

File: pkgs/servers/vouch-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/vouch-proxy/default.nix:12]:
```suggestion
    rev = "v${version}";
```

File: pkgs/servers/vouch-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/vouch-proxy/default.nix:18]: Please add a comment why checks are disabled.

File: pkgs/servers/vouch-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/vouch-proxy/default.nix:35]:
```suggestion
  meta = with.lib; {
```

File: pkgs/servers/vouch-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/servers/vouch-proxy/default.nix:37]: license and maintainer are required.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
> I don't have a lot of time to work on this, so I'm explicitly not stepping up to be a maintainer here. I may not be very responsive to PR comments, either. Apologies in advance.

We cannot accept new packages without maintainers.

> Still, I figure this code drop is better than keeping the work to myself, and so here it is. If someone else wants to take this PR and run with it, that would be much appreciated. Feel free to remix the code as you see fit.

The package itself is very easy and anyone with a bit of nixpkgs experience could reproduce it in a minute. We might as well close this PR if you are not going to complete it.

File: pkgs/servers/vouch-proxy/default.nix
Comment (richardlarocque) [reply pkgs/servers/vouch-proxy/default.nix:18]: I believe the issue is that their tests expect to be invoked by some non-standard wrapper scripts, which the nixpkgs builders knows nothing about.  It probably could be worked around with sufficient effort.

**Comment by richardlarocque** [comment]
Thanks for the review and clear explanation.

I would love to step up to be a maintainer, but that would be making promise I can't keep.

I'm fine with closing this.  I figured this starting point is slightly better than starting from zero, but I'll leave it up to the eventual maintainer to decide.

Thank you for all your hard work on nixpkgs.

**Pull request closed without merging**
